airbyte-integrations/connectors/source-pipedrive/source_pipedrive/streams.py

```diff
@@ -0,0 +1,192 @@
+                the next page in the response.
+                If there are no more pages in the result, return None.
+        """
+        pagination_data = response.json().get("additional_data", {}).get("pagination", {})
+        if pagination_data.get("more_items_in_collection") and pagination_data.get("start") is not None:
+            start = pagination_data.get("start") + self.limit
+            return {"start": start}
+
+    def request_params(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+        params = super().request_params(stream_state=stream_state, stream_slice=stream_slice, next_page_token=next_page_token)
+        next_page_token = next_page_token or {}
+        params.update({"limit": self.limit, **next_page_token})
+        return params
+
+
+class IncrementalPipedriveStream(PaginatedPipedriveStream, ABC):
+    cursor_field = "update_time"
+
+    def __init__(self, replication_start_date: pendulum.datetime, **kwargs):
+        super().__init__(**kwargs)
+        self._replication_start_date = replication_start_date
+
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
+        """
+        Return the latest state by comparing the cursor value in the latest record with the stream's most recent state object
+        and returning an updated state object.
+        """
+        latest_benchmark = latest_record[self.cursor_field]
+        if current_stream_state.get(self.cursor_field):
+            return {self.cursor_field: max(latest_benchmark, current_stream_state[self.cursor_field])}
+        return {self.cursor_field: latest_benchmark}
+
+    @property
+    def path_param(self):
+        return self.name[:-1]
+
+    def request_params(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+        params = super().request_params(stream_state=stream_state, stream_slice=stream_slice, next_page_token=next_page_token)
+        replication_start_date = self._replication_start_date
+        if stream_state.get(self.cursor_field):
+            replication_start_date = max(pendulum.parse(stream_state[self.cursor_field]), replication_start_date)
+        params.update(
+            {
+                "items": self.path_param,
+                "since_timestamp": replication_start_date.strftime("%Y-%m-%d %H:%M:%S"),
+            }
+        )
+        return params
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        """
+        :return an iterable containing each record in the response
+        """
+        records = super().parse_response(response=response, **kwargs)
```

Comment:
please help me to understand, so the incremental streams have double nested records in response?
like 
```json
"data": {
   "data": [
       {...}, {....}, {....}
    ],
}
```

Reply:
yes, exactly